Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py

```diff
@@ -638,6 +638,24 @@ def workflow_(self, workflows, num, fragment, stepList):
     offset = 0.1001,
 )
 
+class UpgradeWorkflow_DD4hep(UpgradeWorkflow):
+    def setup_(self, step, stepName, stepDict, k, properties):
+        stepDict[stepName][k] = merge([{'--geometry': 'DD4hepExtended2021', '--era': 'Run3_dd4hep'}, stepDict[step][k]])
+    def condition(self, fragment, stepList, key, hasHarvest):
+        return '2021' in key
```

Comment:
do we need to add this for every gen fragment, or is ttbar sufficient? (in other words, i recommend adding `fragment=="TTbar_13"` as an additional condition)

Reply:
@kpedro88 well, for initial tests I believe that particle guns would be more suitable, TTbar is a usual reference in the longer term. We may specialize it if really needed, but is this really creating a problem?

Reply:
It just makes the upgrade matrix list of workflows very long.

Reply:
well, in my check il looks like a few percent effect, anyway we may reduce to a subset of reference samples, TTbar_13, SingleMuPt10 and ZMM_13 which was chosen by @slomeo in the test recently added by @bsunanda